src/UI/Component/MainControls/Factory.php

```diff
@@ -0,0 +1,271 @@
+	 * @return  \ILIAS\UI\Component\MainControls\Metabar
+	 */
+	public function metabar(Image $logo): Metabar;
+
+	/**
+	 * ---
+	 * description:
+	 *   purpose: >
+	 *     The Mainbar is a unique page section that bundles access to content-
+	 *     based navigational strategies (like the repository tree) as well as
+	 *     navigation to services unrelated to the actual content, like the
+	 *     administrative settings.
+	 *
+	 *     Since the controls necessary for theses purposes might be quite complex,
+	 *     they are summed up in an easy to grasp Icon or Glyph in conjunction with
+	 *     a short text. Theses reductions form the entries for the mainbar, which
+	 *     thus is the primary list of navigational options for the user and the
+	 *     usual starting point for the user to explore the system.
+	 *
+	 *     There are entries in the bar that are never modified by changing context,
+	 *     but may vary according to e.g. the current user's permissions or settings
+	 *     of the installation. There also is the tools-section of entries in the bar
+	 *     that is used to show tools that are opened on request of the user, e.g. the
+	 *     help, or depending on requirements of the content, e.g. a local navigation.
+	 *
+	 *   composition: >
+	 *     The Mainbar holds Slates and Bulky Buttons.
+	 *
+	 *     In a desktop environment, a vertical bar is rendered on the left side
+	 *     of the screen covering the full height (minus header- and footer area).
+	 *     Entries are aligned vertically.
+	 *
+	 *     In a mobile context, the bar will be rendered horizontally on the bottom.
+	 *
+	 *     When the entries of a Mainbar exceed the available height (mobile: width),
+	 *     remaining buttons will be collected in a "..."-Button.
+	 *
+	 *     The Mainbar is always visible and available (except in specialized views
+	 *     like the exam mode) as a static screen element unaffected by scrolling.
+     *
+	 *   effect: >
+     *     Clicking an entry will carry out its configured action. For slates, this
+	 *     is expanding the slate, while for links an according page opens.
+	 *
+	 *     Buttons in the Mainbar are stateful, i.e. they have a pressed-status
+	 *     that can either be toggled by clicking the same button again or by
+	 *     clicking a different button. This does not apply to Buttons that direcly
+	 *     change the page.
+	 *
+	 *     Opening a slate by clicking an entry will close all other slates in the
+	 *     Mainbar. On desktop, slates open on the right hand of the Mainbar, between
+	 *     bar and content, thus "pushing" the content to the right, if there is not
+	 *     enough room.
+	 *
+	 *     If the content's width would fall below its defined minimum, the expanded
+	 *     slate is opened above (like in overlay, not "on top of") the content.
+	 *
+	 *     The slates height equals that of the Mainbar. Also, their position will
+	 *     remain static when the page is scrolled. A button to close a slate is
```

Comment:
This maybe misread as "position static" in the CSS sense, but I think you mean "fixed" here.